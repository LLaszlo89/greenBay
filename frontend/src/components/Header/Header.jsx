import React from "react";
import { AppBar, Toolbar, Typography } from "@material-ui/core";
import Button from "@material-ui/core/Button";
import CardMedia from "@material-ui/core/CardMedia";
import Grid from "@material-ui/core/Grid";

const Header = () => {
  const name = localStorage.getItem("username");
  const picture = localStorage.getItem("picture");
  const cash = localStorage.getItem("cash_balance");
  
  const handelLogout = () => {
    localStorage.clear();
    window.location.reload();
  };

  const isLoggedIn = name ? (
    <div>
      <AppBar position="static">
        <Toolbar>
          <Grid container spacing={1} justify="space-between">
            <Grid item container direction="row" alignContent="center">
              <CardMedia
                component="img"
                style={{ maxWidth: "50px", borderRadius: "10px" }}
                image={picture}
              />
              <Grid item>
                <Typography variant="h6">{name}</Typography>
                <Typography variant="h6">{cash && cash + "$"} </Typography>
              </Grid>
              <Grid item></Grid>
            </Grid>
          </Grid>

          <Button
            style={{ color: "white" }}
            onClick={() => {
              handelLogout();
            }}
          >
            Logout
          </Button>
        </Toolbar>
      </AppBar>
    </div>
  ) : (
    <div>
      <AppBar position="static">
        <Toolbar>
        </Toolbar>
      </AppBar>
    </div>
  );

  return <div>{ isLoggedIn }</div>;
};

export default Header;
