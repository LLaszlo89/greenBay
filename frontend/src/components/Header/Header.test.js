 import React from "react";
import { shallow } from "enzyme";
import Header from "./Header";

const setUp = (props = {}) => {         // This returns a shallow render
  const component = shallow(<Header {...props} />);
  return component;
};

const findByTestAtr =(component , attr )=>{
const wrapper = component.find(`[data-test = ${attr}]`)
}

describe("Header component", () => {

const dummyData = {username : "TestName", cash:10, pic : "www.testpicture.com"}

  let component;
  beforeEach(() => {      /*  Runs before each test case  /first thing/  */
    component = setUp(dummyData);
  });

  it("Should render without error", () => {   //we can look for classnames but data-test is safer 
    const wrapper = component.find([`[data-test = 'headersComponent']`]);
    expect(wrapper.length).toBe(1);
  });

});
 