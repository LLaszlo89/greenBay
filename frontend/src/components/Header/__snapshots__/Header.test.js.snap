// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Connect component to redux store  Header Component should render correctly 1`] = `
<div>
  <div>
    <header
      className="MuiPaper-root MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorPrimary makeStyles-root-1 MuiPaper-elevation4"
      data-test="headersComponent"
    >
      <div
        className="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
      >
        <div
          className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1 MuiGrid-justify-xs-space-between"
        >
          <div
            className="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-align-content-xs-center"
          >
            <img
              className="MuiCardMedia-root makeStyles-img-3 MuiCardMedia-media MuiCardMedia-img"
              src="Test"
            />
            <div
              className="MuiGrid-root MuiGrid-item"
            >
              <h6
                className="MuiTypography-root MuiTypography-h6"
              >
                Test
              </h6>
              <h6
                className="MuiTypography-root MuiTypography-h6"
              >
                2911 green_$
              </h6>
            </div>
            <div
              className="MuiGrid-root MuiGrid-item"
            />
          </div>
        </div>
        <a
          className="makeStyles-link-4"
          href="/create"
          onClick={[Function]}
        >
          Sale you item
        </a>
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-button-2"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiButton-label"
          >
            Logout
          </span>
        </button>
      </div>
    </header>
  </div>
</div>
`;
